projects/plugins/jetpack/extensions/blocks/donations/amount/index.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { button as icon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import edit from './edit';
+import save from './save';
+
+const name = 'donations-amount';
+const BUTTON_BEHAVIOUR_TITLE = __( 'Amount selector', 'jetpack' );
+const INPUT_BEHAVIOUR_TITLE = __( 'Custom amount field', 'jetpack' );
+
+const settings = {
+	title: __( 'Donations amount button', 'jetpack' ),
+	description: __( 'Button that determines a fixed donation amount', 'jetpack' ),
+	category: 'earn',
+	attributes: {
+		baseAmountMultiplier: {
+			type: 'string',
```

Comment:
Is there a reason why `baseAmountMultiplier` and `amount` are not integers? 🤔

Reply:
You can donate a noninteger amount, for example, $10.50 and we need to preserve the decimal places, 10.00 instead of 10. I guess we could change the baseAmountMultiplier to integer though.

Reply:
Oh, that's true about `amount`. But couldn't we then use `number?

Reply:
- [ ] Update amount and baseAmountMultiplier types to number and integer respectively.